Pseudo-PR: First code review

likelet opened the pull request:
Many thanks to contributing to nf-core/lncpipe!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/lncpipe branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/lncpipe)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [x] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/lncpipe/tree/master/.github/CONTRIBUTING.md

likelet committed:
Merge pull request #7 from nf-core/dockerfile

Functional Dockerfile

drejom committed:
Add per-process resource allocations
likelet/LncPipe#26
and a few other changes detailed there

likelet committed:
Merge pull request #8 from drejom/dev

Add per-process resource allocations

likelet committed:
add plek

likelet committed:
add lncpipereporter and multiqc

likelet committed:
update the font of the logo in Mavon Pro bold

likelet committed:
Update Dockerfile

add email information

likelet committed:
Update Dockerfile

likelet committed:
add workflow

likelet committed:
Simplified the Markdown

likelet committed:
Update README.md

likelet committed:
Update README.md

likelet committed:
Update output.md

likelet committed:
Update output.md

likelet committed:
Update troubleshooting.md

likelet committed:
Update reference_genomes.md

likelet committed:
update cpu profile and test profile

likelet committed:
restructure the code  and add cpat_model file here to avoid the input parameter of cpat_path

likelet committed:
restructure code

likelet committed:
first version that passed the nf-core lint

likelet committed:
remove process resource setting for non-existing process

likelet commented:
BTW, I didn't add test data here, as the test data is extremely big (about 2 Gb).  I created a test data repo on [testdata](https://github.com/likelet/LncPipeTestData) and store the data at a local server that allowing remote accessed. Plz let me know if there is an alternative strategy.

apeltzer commented on .travis.yml:
```diff
@@ -11,9 +11,9 @@ before_install:
   - '[ $TRAVIS_PULL_REQUEST = "false" ] || [ $TRAVIS_BRANCH != "master" ] || ([ $TRAVIS_PULL_REQUEST_SLUG = $TRAVIS_REPO_SLUG ] && [ $TRAVIS_PULL_REQUEST_BRANCH = "dev" ])'
   # Pull the docker image first so the test doesn't wait for this
-  - docker pull nfcore/lncpipe
+  - docker pull likelet/lncpipe
```
```suggestion
  - docker pull nfcore/lncpipe
```

apeltzer replied:
There is a docker image present now there :-)

likelet replied:
bravo

likelet replied:
fixed already

apeltzer commented on .travis.yml:
```diff
@@ -11,9 +11,9 @@ before_install:
+  - docker pull likelet/lncpipe
   # Fake the tag locally so that the pipeline runs properly
-  - docker tag nfcore/lncpipe nfcore/lncpipe:latest
+  - docker tag likelet/lncpipe likelet/lncpipe:latest
```
```suggestion
  - docker tag nfcore/lncpipe nfcore/lncpipe:latest
```

apeltzer commented on conf/awsbatch.config:
```diff
@@ -1,6 +1,6 @@
 /*
  * -------------------------------------------------
- *  Nextflow config file for AWS Batch
+ *  Nextflow config file for AWS Batch [not used for lncpipe]
```
```suggestion
 *  Nextflow config file for AWS Batch
```

Someone might want to use it there, just keep it in ;-)

likelet replied:
Okey

apeltzer commented on conf/base.config:
```diff
@@ -22,14 +22,137 @@ process {
-  withName: fastqc {
-    errorStrategy = { task.exitStatus in [143,137] ? 'retry' : 'ignore' }
+
+  withName:combine_public_annotation {
```
Looks like you have 2-3 configurations in general. You could use the 

`withLabel:highmem` directive to define one resource scope for multiple processes. 

https://www.nextflow.io/docs/latest/config.html#process-selectors

apeltzer commented on conf/igenomes.config:
```diff
@@ -1,10 +1,11 @@
 /*
  * -------------------------------------------------
- *  Nextflow config file for iGenomes paths
+ *  Nextflow config file for iGenomes paths[Not used in lncpipe]
```
In this case, drop the entire file :-)

apeltzer commented on docs/output.md:
```diff
@@ -1,40 +1,72 @@
-## MultiQC
-[MultiQC](http://multiqc.info) is a visualisation tool that generates a single HTML report summarising all samples in your project. Most of the pipeline QC results are visualised in the report and further statistics are available in within the report data directory.
+**Output directory: `results/Identified_lncRNA`**  
+
```
Maybe write a bit more what these files mean?

likelet replied:
Sure

apeltzer reviewed (changes requested):
Some remarks, I'll go over it in more detail later

likelet commented:
> > BTW, I didn't add test data here, as the test data is extremely big (about 2 Gb). I created a test data repo on [testdata](https://github.com/likelet/LncPipeTestData) and stored the data at a local server that allowing remote access. Plz let me know if there is an alternative strategy.
> 
> Hosting somewhere else is fine I guess, but please still add a separate branch for lncpipe and state that in a README in that branch so that people know where the test data resides.
> 
> Do you have an idea how long your tests are running? Or could you strip the dataset down in size while still running proper tests?

Okey, i wll add a branch there. For now , the test data may take less than 10 mins to with default setting . I tried stripping the dataset into chrX only for it can produce meanful result which also can promote the understanding to the result.

likelet committed:
Delete igenomes.config

likelet committed:
Update .travis.yml

likelet committed:
Update awsbatch.config

likelet committed:
Merge branch 'first-commit' into dev

likelet committed:
add mapping file

likelet committed:
add test remotely

likelet committed:
rename main file

likelet committed:
modify travis

likelet committed:
update continaer

likelet committed:
Update .travis.yml

likelet committed:
Update .travis.yml

likelet committed:
Update .travis.yml

likelet committed:
Update .travis.yml

likelet commented:
> > BTW, I didn't add test data here, as the test data is extremely big (about 2 Gb). I created a test data repo on [testdata](https://github.com/likelet/LncPipeTestData) and stored the data at a local server that allowing remote access. Plz let me know if there is an alternative strategy.
> 
> Hosting somewhere else is fine I guess, but please still add a separate branch for lncpipe and state that in a README in that branch so that people know where the test data resides.
> 
> Do you have an idea how long your tests are running? Or could you strip the dataset down in size while still running proper tests?

The test dataset has been trimmed into 0.5Gb in total,  and hosted on my own server. I also set the url in the `.travis.yml `, and make it  pass the CI building. User can also down the data with `wget` command. I will write a detailed document it on this.

likelet committed:
Update Main.nf

likelet committed:
update output document

likelet committed:
update usage document

maxulysse commented on .travis.yml:
```diff
@@ -11,9 +11,9 @@ before_install:
+  - docker pull nfcore/lncpipe:dev
   # Fake the tag locally so that the pipeline runs properly
-  - docker tag nfcore/lncpipe nfcore/lncpipe:latest
+  - docker tag nfcore/lncpipe:dev nfcore/lncpipe:dev
```
I'm not sure that you really need to tag the `dev` as `dev`

likelet replied:
i will update this once released.

maxulysse commented on .travis.yml:
```diff
@@ -27,10 +27,14 @@ install:
 env:
   - NXF_VER='0.32.0' # Specify a minimum NF version that should be tested and work
-  - NXF_VER='' # Plus: get the latest NF version and check that it works
+  # - NXF_VER='' # Plus: get the latest NF version and check that it works
```
I know that tests are longer with that, but, it's quite useful to know if it works for newer versions as well

likelet replied:
ok.

maxulysse commented on Dockerfile:
```diff
@@ -1,135 +1,10 @@
-	rm /opt/sambamba_v0.6.7_linux.tar.bz2
+
+FROM nfcore/base
+MAINTAINER Qi Zhao <zhaoqi@sysucc.org.cn>
```
MAINTAINER is deprecated, you should use LABEL for that instead: cf [Docker docs](https://docs.docker.com/engine/reference/builder/#maintainer-deprecated)

likelet committed:
update docker file  "LABEL maintainer"

maxulysse commented on Dockerfile:
```diff
@@ -1,6 +1,6 @@
 
 FROM nfcore/base
-MAINTAINER Qi Zhao <zhaoqi@sysucc.org.cn>
+LABEL maintainer="zhaoqi@sysucc.org.cn"
```
You can make only one LABEL declaration ;-)

likelet committed:
Update Main.nf

rename the process name that solved the duplicated processed error by updated Nextflow version from 0.18

likelet committed:
Update CHANGELOG.md

ewels committed:
Rename Main.nf to main.nf

mashehu committed:
Update nextflow.config

mashehu committed:
Update nextflow.config

Co-authored-by: Phil Ewels <phil.ewels@scilifelab.se>

likelet committed:
Merge pull request #27 from nf-core/fix-nextflow.config

Update nextflow.config
